src/lazy/text/buffer.rs

```diff
@@ -1987,11 +2042,20 @@ impl<'data> Stream for TextBuffer<'data> {
     }
 
     fn checkpoint(&self) -> Self::Checkpoint {
-        *self
+        let mut checkpoint = *self;
+        // Reset row, column at checkpoint
+        checkpoint.row = 0;
+        checkpoint.prev_newline_offset = 0;
+        checkpoint
     }
 
     fn reset(&mut self, checkpoint: &Self::Checkpoint) {
+        let current_row = self.row;
+        let prev_column_value = self.prev_newline_offset;
+
         *self = *checkpoint;
+        self.row = current_row + checkpoint.row;
+        self.prev_newline_offset = prev_column_value + checkpoint.prev_newline_offset;
```

Comment:
In `TextBuffer`'s case, `Self::Checkpoint` is `TextBuffer`. Saving with `checkpoint` copies `self`, and restoring with `reset` overwrites `self` with `checkpoint`. As a result, I believe that this method should just be `*self = *checkpoint`.

Reply:
I thought the checkpoint was not saving states for row and column values properly. But upon removing my changes for checkpoint and reset like you said it correctly saves the state of self at checkpoint and reset copies the state point checkpoint correctly.

Reply:
I think I found out why I had identified the row and column values were incorrect with checkpoint reset. But if you think I am missing something here do let me know.
Whenever I think the input is taken, I followed the path and seemed to find [this place](https://docs.rs/winnow/latest/src/winnow/combinator/impls.rs.html#399) where it saves the checkpoint before parsing and then updates with reset based on that saved checkpoint. Which means nothing gets updated (offset, row or column). But the offset is explicitly updated with a call to `next_slice` passing the updated offset value after parsing. However, row and column stays the same as prior to parsing which means it doesn't get updated properly unless we add the saved state to the current state like how I do in the `reset` here.

Reply:
`next_slice` only takes the `offset` as input. And this definition is based on [Stream](https://docs.rs/winnow/latest/winnow/stream/trait.Stream.html#tymethod.next_slice) trait. We need to pass the row and column values and thats not possible with this method.

Reply:
Let's create an issue to follow up on this, and move on for now, then.